### ISSUE TITLE
Error from enableHostPolicy 

### ISSUE THREAD

**Issue by seungsoo-lee** [kubearmor/KubeArmor#195]
OS: ubuntu 18.04
Kernel: 4.15

when kubearmor is deployed with enableHostPolicy=true, the below errors pop up and it is not working properly.

```
2021-07-08 06:09:40.191482	INFO	Started to serve gRPC-based log feeds
2021-07-08 06:09:40.501405	INFO	Initializing an eBPF program
2021-07-08 06:10:17.475671	INFO	Initialized the eBPF program
cannot attach kprobe, Resource busy
2021-07-08 06:10:17.866656	ERROR	Failed to initialize the system monitor
github.com/kubearmor/KubeArmor/KubeArmor/log.Err
	/usr/src/KubeArmor/KubeArmor/log/logger.go:99
github.com/kubearmor/KubeArmor/KubeArmor/feeder.(*Feeder).Err
	/usr/src/KubeArmor/KubeArmor/feeder/feeder.go:423
github.com/kubearmor/KubeArmor/KubeArmor/core.KubeArmor
	/usr/src/KubeArmor/KubeArmor/core/kubeArmor.go:354
main.main
	/usr/src/KubeArmor/KubeArmor/main.go:60
runtime.main
	/usr/local/go/src/runtime/proc.go:204
```

**Comment by geyslan** [comment]
Tested on `NETNEXT=1` and went ok.
<details>
<summary>NETNEXT=1</summary>

```bash

vagrant@kubearmor-dev-next:~/KubeArmor/KubeArmor$ git log -n 1
commit f4fba7cf59b62145fffeff5c63d74bd89fbd96b2 (HEAD -> master, upstream/master)
Merge: 323d1eb a5dcd54
Author: Jaehyun Nam <jn@accuknox.com>
Date:   Thu Jul 8 00:31:29 2021 +0900

    Merge pull request #192 from geyslan/feat/makefile/vagrant
    
    makefile: add vagrant rules (status, halt)

vagrant@kubearmor-dev-next:~/KubeArmor/KubeArmor$ uname -a
Linux kubearmor-dev-next 5.13.0-051300rc3-generic #202105232230 SMP Tue Jun 15 12:37:40 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

vagrant@kubearmor-dev-next:~/KubeArmor/KubeArmor$ kubectl proxy &
[1] 8381

vagrant@kubearmor-dev-next:~/KubeArmor/KubeArmor$ Starting to serve on 127.0.0.1:8001

vagrant@kubearmor-dev-next:~/KubeArmor/KubeArmor$ sudo ./kubearmor -enableHostPolicy=true
2021-07-08 11:56:56.068485	INFO	Started to serve gRPC-based log feeds
2021-07-08 11:56:56.069887	INFO	Initializing an eBPF program
2021-07-08 11:57:00.009998	INFO	Initialized the eBPF program
2021-07-08 11:57:00.519473	INFO	Started to monitor system events
2021-07-08 11:57:00.539830	INFO	Registered an AppArmor host profile in kubearmor-dev-next
2021-07-08 11:57:00.539900	INFO	Initialized AppArmor Enforcer
2021-07-08 11:57:00.539909	INFO	Started to protect a host and containers
2021-07-08 11:57:01.541825	INFO	Initialized the Kubernetes client
2021-07-08 11:57:01.541881	INFO	Started to monitor Pod events
2021-07-08 11:57:01.541916	INFO	Started to monitor security policies
2021-07-08 11:57:01.541925	INFO	Started to monitor host security policies
2021-07-08 11:57:01.554233	INFO	Detected a Pod (added/kube-system/kube-scheduler-kubearmor-dev-next)
2021-07-08 11:57:01.554774	INFO	Detected a Pod (added/kube-system/kube-controller-manager-kubearmor-dev-next)
2021-07-08 11:57:01.555170	INFO	Detected a Pod (added/kube-system/kube-proxy-6qr4p)
2021-07-08 11:57:01.555950	INFO	Detected a Pod (added/kube-system/cilium-s5rxl)
2021-07-08 11:57:01.556797	INFO	Container Runtime: docker://18.3.1
2021-07-08 11:57:01.563333	INFO	Detected a container (added/0a30d84cdc5f)
2021-07-08 11:57:01.564072	INFO	Detected a container (added/0106252b6423)
2021-07-08 11:57:01.564587	INFO	Detected a container (added/39cce734e605)
2021-07-08 11:57:01.565194	INFO	Detected a container (added/e816ca1b6cc3)
2021-07-08 11:57:01.566098	INFO	Detected a container (added/603b7bd229cb)
2021-07-08 11:57:01.566757	INFO	Detected a container (added/14d3c50c609e)
2021-07-08 11:57:01.567389	INFO	Detected a container (added/c34a6d21bd7c)
2021-07-08 11:57:01.567764	INFO	Detected a container (added/b3bafafc26cc)
2021-07-08 11:57:01.568673	INFO	Detected a container (added/360351a90c5a)
2021-07-08 11:57:01.569243	INFO	Detected a container (added/23db888cdcf5)
2021-07-08 11:57:01.570157	INFO	Detected a container (added/7068438af404)
2021-07-08 11:57:01.570996	INFO	Detected a container (added/41c725d9d621)
2021-07-08 11:57:01.571741	INFO	Detected a container (added/6473c837a214)
2021-07-08 11:57:01.572533	INFO	Detected a container (added/159ebb345fcb)
2021-07-08 11:57:01.571414	INFO	Detected a Pod (added/kube-system/cilium-operator-86bcb89dc8-fshf6)
2021-07-08 11:57:01.573158	INFO	Detected a container (added/2afe5af5437b)
2021-07-08 11:57:01.574976	INFO	Detected a container (added/0cb1bfdb2b23)
2021-07-08 11:57:01.575628	INFO	Detected a container (added/bb4253c3cc7c)
2021-07-08 11:57:01.576298	INFO	Detected a container (added/2143cc57610b)
2021-07-08 11:57:01.576510	INFO	Started to monitor Docker events
2021-07-08 11:57:01.578553	INFO	Detected a Pod (added/kube-system/coredns-558bd4d5db-jsrmh)
2021-07-08 11:57:01.579072	INFO	Detected a Pod (added/kube-system/etcd-kubearmor-dev-next)
2021-07-08 11:57:01.579607	INFO	Detected a Pod (added/kube-system/kube-apiserver-kubearmor-dev-next)
2021-07-08 11:57:01.581649	INFO	Detected a Pod (added/kube-system/coredns-558bd4d5db-z4m89)
2021-07-08 11:57:02.576779	INFO	Initialized KubeArmor
^C2021-07-08 11:57:24.305783	INFO	Got a signal to terminate the KubeArmor
2021-07-08 11:57:24.337800	INFO	Unegistered an AppArmor host profile in kubearmor-dev-next
2021-07-08 11:57:24.337894	INFO	Destroyed AppArmor Enforcer
2021-07-08 11:57:24.337921	INFO	Stopped the runtime enforcer
2021-07-08 11:57:26.281760	INFO	Stopped the system monitor
2021-07-08 11:57:26.281798	INFO	Terminated the KubeArmor
2021-07-08 11:57:28.283236	INFO	Stopped the log feeder
2021-07-08 11:57:28.283300	INFO	Waiting for remaining routine terminations
```

</details>

Reproducible on `NETNEXT=0`.

<details>
<summary>NETNEXT=0</summary>

```bash
vagrant@kubearmor-dev:~/KubeArmor/KubeArmor$ kubectl proxy &
[1] 20331
vagrant@kubearmor-dev:~/KubeArmor/KubeArmor$ Starting to serve on 127.0.0.1:8001

vagrant@kubearmor-dev:~/KubeArmor/KubeArmor$ enableHostPolicy=true
vagrant@kubearmor-dev:~/KubeArmor/KubeArmor$ sudo ./kubearmor -enableHostPolicy=true
2021-07-08 12:37:58.783449	INFO	Started to serve gRPC-based log feeds
2021-07-08 12:37:58.785550	INFO	Initializing an eBPF program
2021-07-08 12:38:02.250337	INFO	Initialized the eBPF program
cannot attach kprobe, Device or resource busy
2021-07-08 12:38:02.491209	ERROR	Failed to initialize the system monitor
github.com/kubearmor/KubeArmor/KubeArmor/log.Err
	/home/vagrant/KubeArmor/KubeArmor/log/logger.go:99
github.com/kubearmor/KubeArmor/KubeArmor/feeder.(*Feeder).Err
	/home/vagrant/KubeArmor/KubeArmor/feeder/feeder.go:423
github.com/kubearmor/KubeArmor/KubeArmor/core.KubeArmor
	/home/vagrant/KubeArmor/KubeArmor/core/kubeArmor.go:354
main.main
	/home/vagrant/KubeArmor/KubeArmor/main.go:60
runtime.main
	/usr/local/go/src/runtime/proc.go:225
```

**Comment by nam-jaehyun** [comment]
It seems that it happens in older Kernels.
I could see the same error on Ubuntu 18.04, but there was no error on Ubuntu 20.04 and Fedora 30.

**Comment by oneiro-naut** [comment]
@nam-jaehyun I did some research and it seems that it's a bcc issue which isn't resolved. This [comment](https://github.com/iovisor/bcc/issues/1631#issuecomment-373096226) here talks about it. Basically for kernel versions <= 4.17 we use debugfs to attach kprobes. For the newer versions we use perf_event_open API instead. In case of perf_event_open there is no issue when we attach multiple kprobes(or kretprobes) from the same program as that is handled in Kernel(and that's why there is no problem in case of newer kernels). In case of legacy debugfs based kprobe_attach bcc needs to be modified a little bit. In the comment that I have referenced that point was stated but the changes were never done and so the problem is still there for older kernels(which includes v4.15) in bcc. This issue needs a patch in bcc. Some modifications [here](https://github.com/iovisor/bcc/blob/b21bd30d092d57752ce2874f04f739a1f0f42b43/src/cc/libbpf.c#L1007) might fix it.
